client/clientfactory.go

```diff
@@ -221,6 +244,27 @@ func (cf *rpcClientFactory) NewAdminClientWithTimeout(
 	return client
 }
 
+func (cf *rpcClientFactory) NewLocalAdminClientWithTimeout(
```

Comment:
nit: there are a lot of duplicated code in this file. I think there is opportunity to reduce some of the duplications (maybe as a future task).

Reply:
clean up